pstore/api.go

```diff
@@ -317,29 +309,56 @@ func (c *ConsumerWithMetrics) Flush() {
 	c.consumer.Flush()
 }
 
+// Instances that want to know when a batch of records are written to
+// persistent store implement this interface.
+type RecordWriterHook interface {
```

Comment:
The Go convention is to have interface type have a *er name. Why not make this RecordWriter?